website/server/libs/cron.js

```diff
@@ -66,20 +66,37 @@ function grantEndOfTheMonthPerks (user, now) {
 
   if (elapsedMonths > 0) {
     plan.dateUpdated = now;
-    // For every month, inc their "consecutive months" counter. Give perks based on consecutive blocks
-    // If they already got perks for those blocks (eg, 6mo subscription, subscription gifts, etc) - then dec the offset until it hits 0
-    _.defaults(plan.consecutive, {count: 0, offset: 0, trinkets: 0, gemCapExtra: 0});
-
-    for (let i = 0; i < elapsedMonths; i++) {
-      plan.consecutive.count++;
-
-      if (plan.consecutive.offset > 1) {
-        plan.consecutive.offset--;
-      } else if (plan.consecutive.count % 3 === 0) { // every 3 months
-        if (plan.consecutive.offset === 1) plan.consecutive.offset--;
-        plan.consecutive.trinkets++;
-        plan.consecutive.gemCapExtra += 5;
-        if (plan.consecutive.gemCapExtra > 25) plan.consecutive.gemCapExtra = 25; // cap it at 50 (hard 25 limit + extra 25)
+
+    let dateCreatedMoment = moment(plan.dateCreated).startOf('month');
```

Comment:
Thanks for the insight. Is there a reason as to why we can't update dateCreated on a new subscription?

So based on what it looks like, createSubscription isn't getting called on a renewed subscription payment from the source (paypal, stripe, etc). Most likely because the webhook isn't being processed, or isn't even configured. Had createSubscription been called on a renew, the proper amount of perks would have been awarded, as can be proved by a unit test. Thus, I was hoping to use something like dateCreated (and other condition checks) to make sure an active 6 or 12-mo subscription existed.